pennylane/templates/subroutines/approx_time_evolution.py

```diff
@@ -0,0 +1,153 @@
+       terms that are explicitly written in terms of products of Pauli matrices (``PauliX``, ``PauliY``, ``PauliZ``, and ``Identity``).
+       Thus, each term in the Hamiltonian must be expressed this way upon input, or else an error will be raised.
+
+    Args:
+        hamiltonian (.Hamiltonian): The Hamiltonian defining the
+           time-evolution operator. The indices of the observables in the Hamiltonian
+           correspond to the ``wires`` element to which each observable is being applied.
+           The Hamiltonian must be explicitly written
+           in terms of products of Pauli gates (X, Y, Z, and I).
+        time (int or float): The time of evolution, namely the parameter :math:`t` in :math:`e^{- i H t}`.
+        n (int): The number of Trotter steps used when approximating the time-evolution operator.
+
+    Raises:
+        ValueError: if inputs do not have the correct format
+
+    .. UsageDetails::
+
+        The template is used inside a qnode:
+
+        .. code-block:: python
+
+            import pennylane as qml
+            from pennylane.templates import ApproxTimeEvolution
+
+            n_wires = 2
+            wires = range(n_wires)
+
+            dev = qml.device('default.qubit', wires=n_wires)
+
+            coeffs = [1, 1]
+            obs = [qml.PauliX(0), qml.PauliX(1)]
+            hamiltonian = qml.Hamiltonian(coeffs, obs)
+
+            @qml.qnode(dev)
+            def circuit(time):
+                ApproxTimeEvolution(hamiltonian, time, 1)
+                return [qml.expval(qml.PauliZ(wires=i)) for i in wires]
+
+        >>> circuit(1)
+        [-0.41614684 -0.41614684]
+    """
+
+    pauli = {"Identity": "I", "PauliX": "X", "PauliY": "Y", "PauliZ": "Z"}
+
+    if not isinstance(hamiltonian, qml.vqe.vqe.Hamiltonian):
+        raise ValueError(
+            "Hamiltonian must be of type pennylane.Hamiltonian, got {}".format(
+                type(hamiltonian).__name__
+            )
+        )
+
+    if not isinstance(time, (int, float, qml.variable.Variable)):
+        raise ValueError("time must be of type int or float, got {}".format(type(time).__name__))
+
+    if not isinstance(n, (int, qml.variable.Variable)):
+        raise ValueError("n must be of type int, got {}".format(type(n).__name__))
+
+    ###############
+
+    theta = []
+    pauli_words = []
+    wire_index = []
+
+    for i, term in enumerate(hamiltonian.ops):
+
+        word = ""
+
+        try:
+            if isinstance(term.name, str):
+                word = pauli[term.name]
+
+            if isinstance(term.name, list):
+                for j in term.name:
+                    word += pauli[j]
+
+        except KeyError as error:
+            raise ValueError(
+                "hamiltonian must be written in terms of Pauli matrices, got {}".format(error)
+            )
+
+        count = 0
+        for j in list(word):
+            if j == "I":
+                count += 1
+
+        if count != len(word):
```

Comment:
```suggestion
        if word.count("I") != len(word):
```
Just some quality-of-life suggestions that make it easier to follow the logic :slightly_smiling_face:

Reply:
Would also be good to leave a quick comment in the code about what this check is for (skipping adding a term)